.drone.yml

```diff
@@ -1,129 +1,200 @@
-    environment:
-    -  PLUGIN_CUSTOM_DNS=1.1.1.1
-    build_args:
-    - VERSION=${DRONE_TAG}
-    when:
-      branch: [master, release/*]
-      event: tag
+    tag: "${DRONE_TAG}"
+    username:
+      from_secret: docker_username
+  when:
+    event:
+    - tag
+    ref:
+    - refs/head/master
+    - refs/heads/alpha
+    - "refs/heads/release/*"
+    - "refs/tags/*"
 
-  docker-publish-agent:
-    image: plugins/docker
+- name: docker-publish-agent
+  pull: default
+  image: plugins/docker
+  settings:
+    build_args:
+    - "VERSION=${DRONE_TAG}"
+    context: package/
+    custom_dns: 1.1.1.1
     dockerfile: package/Dockerfile.agent
+    password:
+      from_secret: docker_password
     repo: rancher/rancher-agent
-    context: package/
-    tag: ${DRONE_TAG}
-    secrets: [docker_username, docker_password]
-    environment:
-    -  PLUGIN_CUSTOM_DNS=1.1.1.1
-    build_args:
-    - VERSION=${DRONE_TAG}
-    when:
-      branch: [master, release/*]
-      event: tag
+    tag: "${DRONE_TAG}"
+    username:
+      from_secret: docker_username
+  when:
+    event:
+    - tag
+    ref:
+    - refs/head/master
+    - refs/heads/alpha
+    - "refs/heads/release/*"
+    - "refs/tags/*"
 
-  github_binary_prerelease:
-    image: plugins/github-release
-    prerelease: true
-    files:
-     - bin/rancher-*
+- name: github_binary_prerelease
+  pull: default
+  image: plugins/github-release
+  settings:
+    api_key:
+      from_secret: github_token
     checksum:
-     - sha256
-    secrets: [github_token]
-    when:
-      branch: [master, release/*]
-      event: tag
-      ref:
-        include: [ refs/tags/*rc* ]
-
-  github_binary_release:
-    image: plugins/github-release
+    - sha256
     files:
-     - bin/rancher-*
+    - "bin/rancher-*"
+    prerelease: true
+  when:
+    event:
+    - tag
+    ref:
+    - refs/head/master
+    - "refs/heads/release/*"
+    - "refs/tags/*rc*"
+    - "refs/tags/*alpha*"
+
+- name: github_binary_release
+  pull: default
+  image: plugins/github-release
+  settings:
+    api_key:
+      from_secret: github_token
     checksum:
-     - sha256
-    secrets: [github_token]
-    when:
-      branch: [master, release/*]
-      event: tag
-      ref:
-        exclude: [ refs/tags/*rc* ]
+    - sha256
+    files:
+    - "bin/rancher-*"
+  when:
+    event:
+    - tag
+    ref:
+      include:
+      - refs/head/master
+      - "refs/heads/release/*"
+      - "refs/tags/*"
+      exclude:
+      - "refs/tags/*rc*"
+      - "refs/tags/*alpha*"
 
-  chart-promote:
-    image: rancher/dapper:1.11.2
-    commands:
-      - dapper chart/copy
-    volumes:
-      - /var/run/docker.sock:/var/run/docker.sock
-    when:
-      event: deployment
-      environment: promote-stable
+- name: chart-promote
+  pull: default
+  image: rancher/dapper:v0.4.1
+  commands:
+  - dapper chart/copy
+  volumes:
+  - name: docker
+    path: /var/run/docker.sock
+  when:
+    event:
+    - promote
```

Comment:
shouldn't this be a `deployment` event?

Reply:
deployment has been deprecated in favor of the less confusing promote. Based on docs.